dvc/stage.py

```diff
@@ -774,7 +774,11 @@ def _run(self):
             env=fix_env(os.environ),
             executable=executable,
         )
-        p.communicate()
+
+        try:
```

Comment:
This doesn't seem like a right way to do this. What we should do instead is setup a signal handler that would ignore SIGINT until we are done with the process. Note that the process that we are spawning shouldn't inherit that signal handler.

Reply:
Also, need to check what is the situation with that on windows. Not sure how it handles signals there.

Reply:
As far as I understand this. When you send Ctrl-C on Linux the whole process group receives it,  thus we just need to ignore it in DVC itself and wait for the command to process it and exit.

Reply:
Here are a few examples:
CTRL+C:
```
(3.7.0-dvc) ➜  dvc git:(http) ✗ dvc run python sleep.py
Running command:
       python sleep.py
^CTraceback (most recent call last):
 File "sleep.py", line 3, in <module>
   time.sleep(100)
KeyboardInterrupt
ERROR: interrupted by the user
```

kill -SIGINT $(PID)
```
(3.7.0-dvc) ➜  dvc git:(http) ✗ dvc run python sleep.py
Running command:
       python sleep.py
Traceback (most recent call last):
 File "sleep.py", line 3, in <module>
   time.sleep(100)
KeyboardInterrupt
ERROR: failed to run command - stage 'Dvcfile' cmd python sleep.py failed

Having any troubles?. Hit us up at https://dvc.org/support, we are always happy to help!
```

As you can see in the first case, both dvc and the process got SIGINT, and in the second case only the process got it, and so dvc was able to handle it gracefully.